### PR TITLE
ui: refactor breakpoints to fit in an enum

### PR DESCRIPTION
**Pull request by mimecuvalo**
@SomeHats i feel like my naming sucks here. halp.

### Change Type

- [x] `patch` — Bug fix

### Release Notes

- Refactor breakpoints into an enum.

### CODE REVIEW

**Comment by SomeHats** [comment]
I actually quite like the fact that we just give these things number rather than giving them semantic meaning. Each of these has been picked fairly arbitrarily based on what looks good at different screen sizes - they don't correspond to any sort of device.

I could see that being a problem if we want to insert new things part way through the sequence? but then again, that's maybe an issue here - what goes between `XS` and `XXS`? but i'd almost be tempted to go like full chaos-mode with `useBreakpoint(maxScreenWidthPx): boolean` over attempting to give these values meaning.

Not something I feel strongly about though - if you're keen on this change then go for it! I'm stumped for better names though as i say 😅

**Review by SomeHats — APPROVED** [review]

File: packages/tldraw/src/lib/ui/constants.ts [hunk 1]
```diff
@@ -1,2 +1,14 @@
...
+export enum PORTRAIT_BREAKPOINT {
+	ZERO = 0,
+	MOBILE_XXS = 1,
+	MOBILE_XS = 2,
+	MOBILE_SM = 3,
+	MOBILE = 4,
+	TABLET_SM = 5,
+	TABLET = 6,
+	DESKTOP = 7,
+}
```
[the remark this answers is not in the document]
Comment (steveruizok) [reply packages/tldraw/src/lib/ui/constants.ts:14]: Screaming snake case for constants though, right?

https://github.com/tldraw/tldraw/blob/137c7155f1a4936fde4fe2852b80bce46633cd39/packages/editor/src/lib/constants.ts#L83-L86

**Review by steveruizok — APPROVED** [review]

File: packages/tldraw/src/lib/ui/constants.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (SomeHats) [reply packages/tldraw/src/lib/ui/constants.ts:14]: wait aren't all enums constant? oh no!!!!

[the remark this answers is not in the document]
Comment (mimecuvalo) [reply packages/tldraw/src/lib/ui/constants.ts:14]: TIL: "screaming snake case" - there's prbly a good Dall-E picture to be had here and/or a scandavian death-metal band

**Pull request merged**